libraries/AP_HAL_ChibiOS/RCOutput_bdshot.cpp

```diff
@@ -172,12 +194,12 @@ void RCOutput::bdshot_ic_dma_allocate(Shared_DMA *ctx)
             if (group.bdshot.ic_dma_handle[icuch] == ctx && group.bdshot.ic_dma[icuch] == nullptr) {
                 chSysLock();
                 group.bdshot.ic_dma[icuch] = dmaStreamAllocI(group.dma_ch[icuch].stream_id, 10, bdshot_dma_ic_irq_callback, &group);
-                chSysUnlock();
 #if STM32_DMA_SUPPORTS_DMAMUX
                 if (group.bdshot.ic_dma[icuch]) {
                     dmaSetRequestSource(group.bdshot.ic_dma[icuch], group.dma_ch[icuch].channel);
                 }
 #endif
+                chSysUnlock();
```

Comment:
is this change needed?

Reply:
All the others are done from I context, so I believe this is a correct change